### ISSUE TITLE
*GitHub* managment scripts

### ISSUE THREAD

**Issue by amtoine** [amtoine/nu-git-manager#8]
what do you think @melMass about adding some *GitHub* management scripts here? :open_mouth: 

i have my [`gh` module](https://github.com/goatfiles/nu_scripts/blob/nightly/scripts/gh.nu) and that little script i just wrote to help me review `nushell` PRs :yum: 
```bash
def "nu-complete list-repos" [context: string] {
    let user = ($context | str replace 'gh\s*pr\s*open\s*' "" | split row " " | get 0)

    http get ({
        scheme: https,
        username: "",
        host: api.github.com,
        path: $"/orgs/($user)/repos",
        params: {
            sort: updated,
            per_page: 100,
            page: 1
        }
    } | url join)
    | select name description
    | rename value description
}

def "nu-complete gh-status" [] {[
    [value description];

    [failure "The CI does not pass."]
    [pending "The CI is currently running."]
    [success "All the CI jobs have passed."]
]}

def "nu-complete gh-review" [] {[
    [value description];

    [none "No review at all."]
    [changes-requested "There are changes to be applied."]
    [approved "The PR has been approved."]
]}

export def "gh pr open" [
    owner: string
    repo: string@"nu-complete list-repos"
    --draft: bool
    --ready: bool  # has precedence over `--draft`
    --status: string@"nu-complete gh-status"
    --review: string@"nu-complete gh-review"
] {
    let draft = (if $draft or $ready {
        $draft and (not $ready)
    })

    let query = [
        [is pr]
        [is open]
        [draft $draft]
        [status $status]
        [review $review]
    ]

    let url = ({
        scheme: https,
        host: github.com,
        path: $"/($owner)/($repo)/pulls",
        params: {
            q: (
                $query
                | where {|it| $it.1 != null}
                | each { str join "%3A" }
                | str join "+"
            )
        }
    } | url join)

    xdg-open $url
}
```
> **Note**
> - added context-aware completion for `repo`
> - added `--draft` option
> - do not add `draft` to the query if neither `--draft` nor `--ready` are raised

i also have a *GitHub* gist management script [here](https://github.com/goatfiles/nu_scripts/blob/nightly/scripts/gist.nu).

**Comment by melMass** [comment]
I would be for doing this in maybe a second step. First getting most core vcs utilities working, making them cross host etc.. Then we could have host specifics? 
But I agree that integrating with gh at one point can be great.
What do you think?

**Comment by amtoine** [comment]
> I would be for doing this in maybe a second step. First getting most core vcs utilities working, making them cross host etc.. Then we could have host specifics? But I agree that integrating with gh at one point can be great. What do you think?
>
ohh yeah i agree, i just wanted to write it somewhere, not to forget :relieved: 

let's focus on the main module and let's get back to this later :wink:

**Comment by amtoine** [comment]
closed by 5d2075445e0835207d025f766354f27002b442ae
